int32 id
int32[] level
int32[] side_ids
int32 shape_id
